docs/SPIR-V.rst

```diff
@@ -260,6 +260,51 @@ Control flows
 
 [TODO]
 
+Functions
+---------
+
+All functions reachable from the entry-point function will be translated into SPIR-V code. Functions not reachable from the entry-point function will be ignored.
+
+Function parameter
+++++++++++++++++++
+
+For a function ``f`` who has a parameter of type ``T``, the generated SPIR-V signature will use type ``T*`` for the parameter. At every call site of ``f``, additional local variables will be allocated to hold the actual arguments. The local variables are passed in as direct function arguments. For example::
+
+  // HLSL source code
+
+  float4 f(float a, int b) { ... }
+
+  void caller(...) {
+    ...
+    float4 result = f(...);
+    ...
+  }
+
+  // SPIR-V code
+
+                ...
+  %i32PtrType = OpTypePointer Function %int
+  %f32PtrType = OpTypePointer Function %float
+      %fnType = OpTypeFunction %v4float %f32PtrType %i32PtrType
+                ...
+
+           %f = OpFunction %v4float None %fnType
+           %a = OpFunctionParameter %f32PtrType
+           %b = OpFunctionParameter %i32PtrType
+                ...
+
+      %caller = OpFunction ...
+                ...
+     %aAlloca = OpVariable %_ptr_Function_float Function
```

Comment:
`%_ptr_Function_float` is the default name generated from SPIRV-Tools. We should keep it for consistency.